### PR TITLE
fix(streaming): stop windowed HLS sessions crashing on segment reports

### PR DESCRIPTION
**Pull request by arsfeld**
## What broke

Every windowed HLS session (`playlist_mode: :window`, the default for any client that doesn't ask for `:full`, and the fallback whenever media duration is unknown) crashed within roughly 60ms of starting, the moment FFmpeg reported its first finished segment.

## Mechanism

`start_backend/6` wired the transcoder's `on_segments` callback unconditionally, for both `:full` and `:window` sessions. A `:window` session has no `TranscodeWindow` (`window: nil`), but `handle_cast({:segments_ready, ...}, state)` called `TranscodeWindow.mark_ready(state.window, indices)`, which pattern-matches `%TranscodeWindow{}` in its head. A `nil` window raised `FunctionClauseError`, killing the session GenServer. Every later `GET /api/v1/hls/<id>/index.m3u8` then 404'd because the session no longer existed.

## Fix

- `start_backend/6` now only wires `on_segments` when the session is `:full` (mirrors how `grid_aligned` is already conditioned on mode).
- `handle_cast/2` gained a `%{window: nil}` clause ahead of the existing one, as defence in depth against any future caller re-introducing the callback for `:window` sessions.

## How this shipped

The existing unit suite (`test/mydia/streaming/hls_session_segments_test.exs`) only ever drove `:full`-mode sessions through its `Harness`, so nothing exercised a real `{:segments_ready, ...}` cast against a windowed session. It was caught by `CI / Player E2E` on master (run 33452019208), not by the unit suite: a session would start, log `Terminating HLS session <id>, reason: {:function_clause, ...}` about 60ms later, and `player/integration_test/p2p_streaming_test.dart` would fail on "HLS playlist should be ready with segments".

Added a regression test that constructs a windowed session state (`window: nil`, `segment_plan: nil`, `playlist_mode: :window`) and sends it `{:segments_ready, generation, [0]}`, asserting the process stays alive. Verified it fails with the exact `FunctionClauseError` when the `%{window: nil}` guard clause is removed, and passes with it restored.

## Test plan

- [x] `./dev mix test test/mydia/streaming/ test/mydia_web/controllers/api/` - 524 tests, 0 failures
- [x] `./dev mix credo --strict` - no issues
- [x] Confirmed the new regression test fails with `FunctionClauseError` when the fix's guard clause is removed

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved HLS streaming session stability when segment notifications are received unexpectedly.
  * Prevented windowed playback sessions from crashing under edge-case timing conditions.
  * Improved playlist mode and segment processing for more reliable playback.

* **Tests**
  * Added regression coverage confirming windowed sessions remain active during unexpected segment notifications.

### CODE REVIEW

File: test/mydia/streaming/hls_session_segments_test.exs [hunk 1]
```diff
@@ -375,4 +375,45 @@ defmodule Mydia.Streaming.HlsSessionSegmentsTest do
...
+      {:ok, pid} = Harness.start_link(state)
+      Process.flag(:trap_exit, true)
+
+      # window_generation defaults to 0 in base_state/1; matching it here
+      # means this notification is not discarded by the stale-generation
+      # guard clause and actually reaches the window: nil clause under test.
+      HlsSession.notify_segments(pid, 0, [0])
+
+      refute_receive {:EXIT, ^pid, _reason}, 200
+      assert Process.alive?(pid)
```
[the remark this answers is not in the document]
Comment (arsfeld) [reply test/mydia/streaming/hls_session_segments_test.exs]: Agreed, applied in d5d57819f.

Replaced both assertions with `assert %{window: nil} = :sys.get_state(pid)`. The old pair was a 200ms sleep dressed as an assertion: `refute_receive` succeeds precisely when nothing arrives, so an unprocessed cast greened the test without ever reaching the clause under test. `:sys.get_state/1` is a system message, and message order from one process to another is guaranteed, so issuing it right after the cast from the same process forces the cast to be handled first; a crash then surfaces as an exit with `:noproc` rather than a silent pass.

Re-verified the test still fails with the same `FunctionClauseError` on `TranscodeWindow.mark_ready/2` when the `%{window: nil}` clause is removed, so the new form is not quietly narrower. It now fails immediately instead of 200ms later.
